File: howooyeon/dfs_bfs/네트워크.java
class Solution {
  public int solution(int n, int[][] computers) {
      boolean[] visited = new boolean[n];
      int count = 0;
      
      for (int i = 0; i < n; i++) {
          if (!visited[i]) {
              dfs(i, computers, visited);
              count++;
          }
      }
      
      return count;
  }
  
  private void dfs(int node, int[][] computers, boolean[] visited) {
      visited[node] = true;
      for (int i = 0; i < computers.length; i++) {
          if (computers[node][i] == 1 && !visited[i]) {
              dfs(i, computers, visited);
          }
      }
  }
}

File: minseong/dfs_bfs/네트워크.js
function solution(n, computers) {
  let visited = new Array(n).fill(false);
  let networkCount = 0;

  function dfs(node) {
    visited[node] = true;
    for (let next = 0; next < n; next++) {
      if (computers[node][next] === 1 && !visited[next]) {
        dfs(next);
      }
    }
  }

  for (let i = 0; i < n; i++) {
    if (!visited[i]) {
      dfs(i);
      networkCount++;
    }
  }

  return networkCount;
}
